feat(studio): usage area brush up

kemaldotearth opened the pull request:
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

## What kind of change does this PR introduce?

This introduces a lot of nit picky fixes and improvements to the org Usage area. The changes will aim to be small in nature and give us a platform to build up and improve this area.

kemaldotearth committed:
fix: date range picker dropdown width

This fixes the date range pickers width on the Usage section. No more crumpled up menu items.

kemaldotearth committed:
chore: remove AI slop empty classes

kemaldotearth committed:
feat: get rid of legacy scaffold header on usage page

kemaldotearth committed:
chore: remove legacy scaffold from no permissions layout on usage

kemaldotearth committed:
feat: improve styling for sticky header area

kemaldotearth committed:
nit: small uppercase for a p

kemaldotearth committed:
fix: padding for no permission banner

Was missing the ScaffoldHeader wrapper

kemaldotearth committed:
feat: update loading skeleton for header area

kemaldotearth committed:
style: same breakpoints for header loading skeletons

kemaldotearth committed:
feat: remove ListBox in favour or Select_Shadcn_ for our project dropdown

kemaldotearth committed:
fix: logic for admonition when project is selected

kemaldotearth committed:
fix: remove rogue border from total usage grid

kemaldotearth committed:
fix: use proper background colour for sticky bar

kemaldotearth committed:
Merge remote-tracking branch 'origin/master' into feat/usage-area-brushup

kemaldotearth committed:
fix: more info link click area

kemaldotearth committed:
fix: reduce width of hover card on compute metrics

kemaldotearth committed:
feat: remove separator from usage section headers

kemaldotearth committed:
fix: revert width of hover cards for compute and billing metric

dnywh committed:
fix type size and chevron tint

dnywh committed:
only wrap anchored metrics in Link

otherwise non-anchored metrics get #undefined Link and unnecessary pointer

kemaldotearth committed:
Merge remote-tracking branch 'origin/master' into feat/usage-area-brushup

kemaldotearth committed:
Merge remote-tracking branch 'origin/master' into feat/usage-area-brushup

kemaldotearth committed:
style: shore up sections with headers

kemaldotearth committed:
Merge remote-tracking branch 'origin/master' into feat/usage-area-brushup

joshenlim commented on apps/studio/components/interfaces/Organization/BillingSettings/BillingBreakdown/ComputeMetric.tsx:
```diff
@@ -55,7 +55,7 @@ const ComputeMetric = ({ slug, metric, usage, relativeToSubscription }: ComputeM
         </div>
       </HoverCardTrigger>
-      <HoverCardContent side="bottom" align="center" className="w-[500px]" animate="slide-in">
+      <HoverCardContent side="bottom" align="end" className="w-[500px]" animate="slide-in">
```
Ohhh its not - but reckon we should make the same change in BillingMetric then?

joshenlim replied:
was the intention to center the hovercard? what about just aligning start?
<img width="610" height="177" alt="image" src="https://github.com/user-attachments/assets/b962bed8-f445-4eef-a600-9f0312ec80f1" />

side note - the width seems a little long for the amount of content in it

kemaldotearth committed:
fix: reset select state to undefined

The setSelectedProjectRef was causing our rings not to show on the usage table.

joshenlim committed:
Fixy

kemaldotearth committed:
Merge remote-tracking branch 'origin/master' into feat/usage-area-brushup

joshenlim commented:
Just a feedback: with the removal of the divider, feels like the sections are a bit more disjointed imo

e.g MAU and MA-SSO users are meant to be under "Activity" but it now seems like only MAU is under "Activity" and MA-SSO users is a separate section
<img width="1446" height="710" alt="image" src="https://github.com/user-attachments/assets/c97e926c-7dd7-45e8-af2a-4a7c2c89627e" />

joshenlim committed:
Nit refactors and fixes

joshenlim reviewed (approved):
Generally LGTM - checked on staging preview but JFYI i'm not able to test `ComputeMetric.tsx` as I don't seem to have that data on my staging organization

Non-blocking feedback RE the removal of the dividers for the SectionHeaders [here](https://github.com/supabase/supabase/pull/38408#issuecomment-3259287832)

dnywh commented on apps/studio/components/interfaces/Organization/BillingSettings/BillingBreakdown/BillingMetric.tsx:
Two minor changes I made to this `BillingMetric` component in this PR:
- https://github.com/supabase/supabase/pull/38408/commits/dd52f8d35dab7deba0a0c98724e47155d963c319: minor type size and icon colour tweak
- https://github.com/supabase/supabase/pull/38408/commits/4aff4a53684e881bff8b04f25bdfc086884081de: moved where the wrapping Link was positioned to avoid `/#undefined` when it isn’t applicable (i.e. that `metric.anchor` ternary update)

kemaldotearth committed:
Merge remote-tracking branch 'origin/master' into feat/usage-area-brushup

kemaldotearth committed:
Merge branch 'feat/usage-area-brushup' of github.com:supabase/supabase into feat/usage-area-brushup
